lua/lspconfig.lua

```diff
@@ -7,41 +7,51 @@ local M = {
 ---@class Alias
 ---@field to string The new name of the server
 ---@field version string The version that the alias will be removed in
----@param name string
----@return Alias
-local function server_alias(name)
-  local aliases = {
-    ['fennel-ls'] = {
-      to = 'fennel_ls',
-      version = '0.2.0',
-    },
-    ruby_ls = {
-      to = 'ruby_lsp',
-      version = '0.2.0',
-    },
-    ['starlark-rust'] = {
-      to = 'starlark_rust',
-      version = '0.2.0',
-    },
-    sumneko_lua = {
-      to = 'lua_ls',
-      version = '0.2.0',
-    },
-    tsserver = {
-      to = 'ts_ls',
-      version = '0.2.0',
-    },
-  }
+---@field inconfig? boolean need shown in lspinfo
+local aliases = {
+  ['fennel-ls'] = {
+    to = 'fennel_ls',
+    version = '0.2.1',
+  },
+  ruby_ls = {
+    to = 'ruby_lsp',
+    version = '0.2.1',
+  },
+  ['starlark-rust'] = {
+    to = 'starlark_rust',
+    version = '0.2.1',
+  },
+  sumneko_lua = {
+    to = 'lua_ls',
+    version = '0.2.1',
+  },
+  tsserver = {
+    to = 'ts_ls',
+    version = '0.2.1',
+  },
+}
 
-  return aliases[name]
+---@return Alias
+M.server_aliases = function(name)
+  if name then
+    return aliases[name]
+  end
+  local result = {}
+  for server_name, deprecated in pairs(aliases) do
+    if deprecated.inconfig then
+      result[server_name] = deprecated
+    end
+  end
+  return result
```

Comment:
didn't test, but something like this should work:

```suggestion
return vim.tbl_filter(function(v) return not not v.inconfig end, aliases)
```

Reply:
vim.iter is suggested ?

Reply:
I wasn't sure if lspconfig requires Nvim 0.10 now. But also, if there aren't multiple stages in the "pipeline", it's fine to use tbl_filter

Reply:
this is a map-like table not list-like table. tbl_filter will return a list-like table we need servername

Reply:
```lua
function vim.tbl_filter(func, t)
  vim.validate({ func = { func, 'c' }, t = { t, 't' } })
  --- @cast t table<any,any>

  local rettab = {} --- @type table<any,any>
  for _, entry in pairs(t) do
    if func(entry) then
      rettab[#rettab + 1] = entry
    end
  end
  return rettab
end

```

vim.tbl_filter just return a list-like table with value in before `t` . then the server name as k will be ignored.